hyperspyui/mainwindowutillayer.py

```diff
@@ -337,6 +339,11 @@ def get_figure_filepath_suggestion(self, figure, deault_ext=None):
         base, tail = os.path.split(f)
         fn, ext = os.path.splitext(tail)
 
+        # Clean filename of any illegal characters:
+        fn = "".join([c for c in fn if
+                      c.isalpha() or c.isdigit() or c == ' '
+                      or c == '_' or c == '-']).rstrip()
```

Comment:
This is very restrictive, so I would at least ensure that this is not run on the complete `fn` (which might include "C:\myfoldername\myfile", in which case the colon in "C:\" would be stripped). I would put this on `f` above:

```python
f = ...
if f:
    f ="".join(...)
else:
    f = self.cur_dir   # Safe to assume this is OK I think
```

Reply:
Actually, never mind that, since we use `split` and `splitext`.

Reply:
Maybe put it behind a `if f == self.cur_dir:` check?

Reply:
I think that shouldn't be necessary because of the `os.path.split`, right?